src/dev/build/tasks/nodejs_modules/code_parser/strategies.js

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { canRequire } from './can_require';
+import { dependenciesVisitorsGenerator } from './visitors';
+import { dirname, isAbsolute } from 'path';
+
+function calculateTopLevelDependency(inputDep, outputDep = '') {
+  // The path separator will be always the forward slash
+  // as at this point we only have the found entries into
+  // the provided source code entries where we just use it
+  const pathSeparator = '/';
+  const depSplitPaths = inputDep.split(pathSeparator);
+  const firstPart = depSplitPaths.shift();
+  const outputDepFirstArgAppend = outputDep ? pathSeparator : '';
+
+  outputDep += `${outputDepFirstArgAppend}${firstPart}`;
+
+  // In case our dependency isn't started by @
+  // we are already done and we can return the
+  // dependency value we already have
+  if (firstPart.charAt(0) !== '@') {
+    return outputDep;
+  }
+
+  // Otherwise we need to keep constructing the dependency
+  // value because dependencies starting with @ points to
+  // folders of dependencies. For example, in case we found
+  // dependencies values with '@the-deps/a' and '@the-deps/a/b'
+  // we don't want to map it to '@the-deps' but also to @'the-deps/a'
+  // because inside '@the-deps' we can also have '@the-dep/b'
+  return calculateTopLevelDependency(depSplitPaths.join(pathSeparator), outputDep);
+}
+
+export async function dependenciesParseStrategy(build, parseSingleFile, mainEntry, wasParsed, results) {
+  // Retrieve native nodeJS modules
+  const natives = process.binding('natives');
+
+  // Get dependencies from a single file and filter
+  // out node native modules from the result
+  const dependencies = (await parseSingleFile(mainEntry, dependenciesVisitorsGenerator))
+    .filter(dep => !natives[dep]);
+
+  // Return the list of all the new entries found into
+  // the current mainEntry that we could use to look for
+  // new dependencies
+  return dependencies.reduce((filteredEntries, entry) => {
+    const absEntryPath = build.resolvePath(dirname(mainEntry), entry);
+    const requiredPath = canRequire(build, absEntryPath);
+    const relativeFile = !isAbsolute(entry);
+    const requiredRelativePath = canRequire(build, entry);
```

Comment:
nit: for consistency with naming below, maybe rename these: `isRequiredPath`, `isRelativeFile`, `isRequiredRelativePath`

Reply:
Thanks for the feedback Josh. I will just change it for `relativeFile` as the 2 others holds strings with paths 😃